examples/api/test/widgets/scroll_view/custom_scroll_view.1_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/scroll_view/custom_scroll_view.1.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('What should be visible or not visible in the initial state.', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    expect(find.byType(Scaffold), findsOne);
+    expect(find.descendant(
+      of: find.byType(Scaffold),
+      matching: find.byType(AppBar),
+    ), findsOne);
+    expect(find.descendant(
+      of: find.byType(AppBar),
+      matching: find.byType(IconButton),
+    ), findsOne);
+    expect(find.descendant(
+      of: find.byType(Scaffold),
+      matching: find.byType(CustomScrollView),
+    ), findsOne);
+    expect(find.descendant(
+      of: find.byType(CustomScrollView),
+      matching: find.byType(SliverList),
+    ), findsOne);
+    expect(find.byType(SliverList), findsOne);
+    // Initial state should present only "Item: 0" on the SliverList.
+    expect(find.widgetWithText(SliverList, 'Item: 0'), findsOne);
+    expect(find.widgetWithText(SliverList, 'Item: -1'), findsNothing);
+    expect(find.widgetWithText(SliverList, 'Item: 1'), findsNothing);
+});
+
+  testWidgets('IconButton click and scroll adds elements to the screen.', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    await tester.tap(find.byType(IconButton));
+    await tester.pump();
+
```

Comment:
Before scrolling, we should check if 'item 1' is visible and 'item -1' not visible.